spec/pluralrules.html

```diff
@@ -75,77 +55,91 @@
-           [[NumberOfFractionDigitsWithoutTrailing]]: _w_,
-           [[FractionDigits]]: _f_,
-           [[FractionDigitsWithoutTrailing]]: _t_
-         }.
-         1. Return _result_.
+         1. Else,
+           1. Let _w_ be 0.
+           1. Let _t_ be 0.
+         1. Return a new Record { [[Number]]: _n_, [[IntegerDigits]]: _i_, [[NumberOfFractionDigits]]: _v_, [[NumberOfFractionDigitsWithoutTrailing]]: _w_, [[FractionDigits]]: _f_, [[FractionDigitsWithoutTrailing]]: _t_ }.
        </emu-alg>
 
        <emu-table id="table-plural-operands">
-         <emu-caption>Plural Rules Operands</emu-caption>
+         <emu-caption>Internal Slots of Plural Rules Operands</emu-caption>
          <table class="real-table">
            <thead>
              <tr>
-               <th>Symbol</th>
-               <th>Value</th>
+               <th>Internal Slot</th>
+               <th>Type</th>
+               <th>Description</th>
              </tr>
            </thead>
            <tr>
-             <td>n</td>
+             <td>[[Number]]</td>
+             <td>Number</td>
              <td>absolute value of the source number (integer and decimals)</td>
            </tr>
            <tr>
-             <td>i</td>
-             <td>integer digits of n.</td>
+             <td>[[IntegerDigits]]</td>
+             <td>Number</td>
+             <td>Number of digits of [[Number]].</td>
            </tr>
            <tr>
-             <td>v</td>
-             <td>number of visible fraction digits in n, <i>with</i> trailing zeros.</td>
+             <td>[[NumberOfFractionDigits]]</td>
+             <td>Number</td>
+             <td>Number of visible fraction digits in [[Number]], <i>with</i> trailing zeros.</td>
            </tr>
            <tr>
-             <td>w</td>
-             <td>number of visible fraction digits in n, <i>without</i> trailing zeros.</td>
+             <td>[[NumberOfFractionDigitsWithoutTrailing]]</td>
+             <td>Number</td>
+             <td>Number of visible fraction digits in [[Number]], <i>without</i> trailing zeros.</td>
            </tr>
            <tr>
-             <td>f</td>
-             <td>visible fractional digits in n, <i>with</i> trailing zeros.</td>
+             <td>[[FractionDigits]]</td>
+             <td>Number</td>
+             <td>Number of visible fractional digits in [[Number]], <i>with</i> trailing zeros.</td>
            </tr>
            <tr>
-             <td>t</td>
-             <td>visible fractional digits in n, <i>without</i> trailing zeros. </td>
+             <td>[[FractionDigitsWithoutTrailing]]</td>
+             <td>Number</td>
+             <td>Number of visible fractional digits in [[Number]], <i>without</i> trailing zeros. </td>
            </tr>
          </table>
        </emu-table>
 
     </emu-clause>
 
+    <emu-clause id="sec-pluralruleselection" aoid="PluralRuleSelection">
+
+      <h1>PluralRuleSelection (locale, type, n, operands)</h1>
```

Comment:
this is the big change here, we use PluralRuleSelection as a hook for implements to do their own magic via implementation defined rules, in any form they want, by providing all the pieces needed to select the right plural form.